File: spl-net/src/main/java/bgu/spl171/net/impl/TFTP/CommandBuilders/ReadRequestCommandBuilder.java
package bgu.spl171.net.impl.TFTP.CommandBuilders;

import bgu.spl171.net.impl.TFTP.Commands.Command;
import bgu.spl171.net.impl.TFTP.Commands.ReadRequest;
import bgu.spl171.net.impl.TFTP.EncoderDecoders.StringEncoderDecoder;

import java.util.ArrayList;
import java.util.List;

public class ReadRequestCommandBuilder implements CommandBuilder {

    StringEncoderDecoder stringEncoderDecoder = new StringEncoderDecoder();

    @Override
    public Command decodeNextByte(byte nextByte) {
        String fileName = stringEncoderDecoder.decodeNextByte(nextByte);
        if(fileName!=null){
            return new ReadRequest(fileName);
        }
        return null;
    }
}

File: java to c++ builders/ReadRequestCommandBuilder.cpp
#include "ReadRequestCommandBuilder.h"

namespace bgu
{
	namespace spl171
	{
		namespace net
		{
			namespace impl
			{
				namespace TFTP
				{
					namespace CommandBuilders
					{
						using bgu::spl171::net::impl::TFTP::Commands::Command;
						using bgu::spl171::net::impl::TFTP::Commands::ReadRequest;
						using bgu::spl171::net::impl::TFTP::EncoderDecoders::StringEncoderDecoder;

						Command *ReadRequestCommandBuilder::decodeNextByte(char nextByte)
						{
							std::wstring fileName = stringEncoderDecoder->decodeNextByte(nextByte);
							if (fileName != L"")
							{
								return new ReadRequest(fileName);
							}
							return nullptr;
						}
					}
				}
			}
		}
	}
}
